pacman/model/graphs/application/application_edge.py

```diff
@@ -61,27 +72,35 @@ def __init__(
         self._pre_vertex = pre_vertex
         self._post_vertex = post_vertex
         self._traffic_type = traffic_type
+        if machine_edge_type is None:
+            machine_edge_type = _machine_edge_class()
```

Comment:
This would be just `MachineEdge` (i.e., effectively a constant as it is a class reference) except I want the resolution to be done at a later time. I put that lookup in its own function simply because it is a nasty bit of ghastliness and so should be corralled on its lonesome.